src/second_ode_solve.jl

```diff
@@ -0,0 +1,63 @@
+struct NNODE2{C,O} <: NeuralNetDiffEqAlgorithm
+    chain::C
+    opt::O
+end
+NNODE2(chain;opt=Flux.ADAM(0.1)) = NNODE2(chain,opt)
+
+function DiffEqBase.solve(
+    prob::DiffEqBase.AbstractODEProblem,
+    alg::NeuralNetDiffEqAlgorithm,
+    args...;
+    dt,
+    timeseries_errors = true,
+    save_everystep=true,
+    adaptive=false,
+    abstol = 1f-6,
+    verbose = false,
+    maxiters = 100)
+
+    DiffEqBase.isinplace(prob) && error("Only out-of-place methods are allowed!")
+
+    u0, du0 = prob.u0
+    tspan = prob.tspan
+    f = prob.f
+    p = prob.p
+
+    #hidden layer
+    chain  = alg.chain
+    opt    = alg.opt
+    ps     = Flux.params(chain)
+    data   = Iterators.repeated((), maxiters)
+
+    #train points generation
+    ts = []
+    cnt = 0.0f0
+    for ii in 1:Int((tspan[2]-tspan[1])/dt[1]+1)
+        push!(ts, cnt)
+        cnt += dt[1]
+    end
+    
+    #Initial Value Problem
+    phi(t) = (u0 .+ du0 .* (t .- tspan[1]) .+ (t .- tspan[1]) .^ 2 .* chain([t]))[1]
+
+
+    dfdx = t -> (phi(t+sqrt(eps(typeof(dt[1])))) - phi(t)) / sqrt(eps(typeof(dt[1])))
+    epsilon(t) = cbrt(eps(real(t)))
```

Comment:
this is going to cause problems at zero. don't make it a function of `t`.

Reply:
Declared t0 and changed to real(t-t0)

Reply:
t0 is a float though, so even if t = 0, wouldn't this still be a float?

Reply:
```julia
julia> eps(0.0)
5.0e-324
```

is a really bad idea for numerical error.

Reply:
Oic! Perhaps a function of dt?

Reply:
This is why I suggested it be based on `eps(Float32)` which is based on 1 since what it needs to be based on is the relative scale of phi(t) which is unknown until it's computed. The theory for `cbrt` is on the arbitrary size of that

Reply:
Ah! At first i didn't really understand eps(Float32)